README.md

```diff
@@ -64,19 +69,159 @@ const main = defineCommand({
       description: "Use friendly greeting",
     },
   },
-  run({ args }) {
+  run({ args }) { // Command can be async
     console.log(`${args.friendly ? "Hi" : "Greetings"} ${args.name}!`);
   },
 });
 
 runMain(main);
 ```
 
+### Sub Commands
+
+You can define sub commands and attach them to main command to create a nested command structure. This is recursive so you can attach sub commands to sub commands, etc.
+
+```js
+import { defineCommand, runMain } from "citty";
+
+// First, you define a new command
+const sub = defineCommand({
+  meta: {
+    name: "sub",
+    description: "Sub command",
+  },
+  args: { // Sub commands can have their own arguments like any other command
+    name: {
+      type: "positional",
+      description: "Your name",
+      required: true,
+    },
+  },
+  run({ args }) { // Command can be async
+    console.log(`Hello ${args.name}!`);
+  },
+});
+
+// Then, you define a main command and attach sub command to it
+const main = defineCommand({
+  meta: {
+    name: "hello",
+    version: "1.0.0",
+    description: "My Awesome CLI App",
+  },
+  commands: {
+    sub, // Attach sub command to main command
+  },
+});
+
+runMain(main);
+```
+
```

Comment:
Adding example of how to execute subcommands from CLI will be much friendly like:

You can now run the sub command that logs your name via:
```bash
node script.ts sub john
// output: Hello john
```

Reply:
Yes, great idea!